### ISSUE TITLE
Cannot initialize my own reformer model from file

### ISSUE THREAD

**Issue by alanyeung95** [google/trax#1621]
### Description

Hi guys, may need some help here, as I has been trying on this for many days and cannot get it solved.

Is it just the problem of the package?

when I run this line of code
```
model.init_from_file('./model/model.pkl.gz',
                     weights_only=True, input_signature=shape11)
```
this error happen:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-40-60e56f7f7cb2> in <module>
      1 # initialize from file
      2 model.init_from_file('./model/model.pkl.gz',
----> 3                      weights_only=True, input_signature=shape11)
      4 
      5 # save the starting state
...
...

~/Git/nlp/lib/python3.7/site-packages/trax/fastmath/jax.py in tree_unflatten(flat, tree, copy_from_tree)
    290       new_tree[k] = new_v
    291     return new_tree, rest
--> 292   return flat[0], flat[1:]
    293 
    294 

IndexError: list index out of range
```

### Environment information

```
OS: Ubuntu 20.04

$ pip freeze | grep trax
trax==1.3.4

$ pip freeze | grep tensor
mesh-tensorflow==0.1.19
tensor2tensor==1.15.7
tensorboard==2.5.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.0
tensorflow==2.4.1
tensorflow-addons==0.12.1
tensorflow-datasets==4.3.0
tensorflow-estimator==2.4.0
tensorflow-gan==2.0.0
tensorflow-hub==0.12.0
tensorflow-metadata==0.30.0
tensorflow-probability==0.7.0
tensorflow-text==2.4.3

$ pip freeze | grep jax
jax==0.1.73
jaxlib==0.1.52


$ python -V
Python 3.7.10
```

### For bugs: reproduction and error logs

```
# Steps to reproduce:
...
```

```
# Error logs:
...
```

**Comment by mansijoshii** [comment]
Hi, what helped resolve the error?

**Comment by alanyeung95** [comment]
@mansijoshii , well in my case, it is because the parameters in my trained model is mismatch with my parameters defined in my reformer/transformer model. 

To be more specific, I defined `n_layers=6` in my trax reformer, however I set `n_layers=2` during my training task, which I believe that is the root cause.

To sum up, it is just a coding issue or simply wrong usage of library instead of a problem of trax, I just wish I realize it earlier.
